main.go

```diff
@@ -57,7 +57,13 @@ func (c *component) New(ctx context.Context, conf *config) (func(context.Context
 		LogFn:           domain.LoggerFromContext,
 		StatFn:          domain.StatFromContext,
 		AssetAttributor: a,
-		Producer:        p,
+		// AttributionFailureHandler field is simply a stub, it varies user to user
```

Comment:
Suggestion: I would update the comment to point out that the implementation will likely vary more from organization to organization than from user to user.

For example, one organization may want to store their attribution failures encrypted in a persistent store, while others may want to persist theirs in a long-lived encrypted queue, or rely on a streaming platform like Kafka to keep the data until it can be investigated.